subsys/bluetooth/controller/ll_sw/ull_llcp.c

```diff
@@ -92,6 +92,22 @@ void llcp_proc_ctx_release(struct proc_ctx *ctx)
 }
 
 #if defined(LLCP_TX_CTRL_BUF_QUEUE_ENABLE)
+/*
+ * @brief Update 'global' tx buffer allowance
+ */
+void ull_cp_update_tx_buffer_queue(struct ll_conn *conn)
+{
+#if (CONFIG_BT_CTLR_LLCP_PER_CONN_TX_CTRL_BUF_NUM > 0)
+	if (conn->llcp.tx_buffer_alloc > CONFIG_BT_CTLR_LLCP_PER_CONN_TX_CTRL_BUF_NUM) {
+		common_tx_buffer_alloc -= (conn->llcp.tx_buffer_alloc -
+					   CONFIG_BT_CTLR_LLCP_PER_CONN_TX_CTRL_BUF_NUM);
+	}
+#else /* CONFIG_BT_CTLR_LLCP_PER_CONN_TX_CTRL_BUF_NUM > 0 */
+	common_tx_buffer_alloc -= conn->llcp.tx_buffer_alloc;
+#endif /* CONFIG_BT_CTLR_LLCP_PER_CONN_TX_CTRL_BUF_NUM > 0 */
```

Comment:
I think we can get rid of the conditional compile as follows:
```suggestion
	if (conn->llcp.tx_buffer_alloc > CONFIG_BT_CTLR_LLCP_PER_CONN_TX_CTRL_BUF_NUM) {
		common_tx_buffer_alloc -= (conn->llcp.tx_buffer_alloc -
					   CONFIG_BT_CTLR_LLCP_PER_CONN_TX_CTRL_BUF_NUM);
```
when CONFIG_BT_CTLR_LLCP_PER_CONN_TX_CTRL_BUF_NUM > 0 this is the code in the '#if'-branch

when CONFIG_BT_CTLR_LLCP_PER_CONN_TX_CTRL_BUF_NUM = 0 the 'if' statement is always true and the code is equivalend to the '#else'-branch

Reply:
Agree - thanks for noticing